src/apirequest.ts

```diff
@@ -181,6 +181,7 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
       const finale = `--${boundary}--`;
       const rStream = new stream.PassThrough();
       const pStream = new ProgressStream();
+      const bStream = new BoundaryStream(finale);
```

Comment:
`PassThrough` streams accept a `flush` function similarly to `Transform` streams. I think you could simplify the whole thing by just providing a `flush` function to `rStream`.

```js
const flush = callback => callback(null, `\r\n${finale}`);
const rStream = new stream.PassThrough({flush});
```